inbox.js

```diff
@@ -20,11 +20,38 @@ function refresh_inbox(wallet) {
 			};
 
     	const res = await this.arweave.api.post(`arql`, query);
+		var mail_UnixTime = [];
 		if(res.data == "") {
 			var mail_ids = [];
 		}
 		else {
 			var mail_ids = res.data;
+
+
+			mail_ids.map(async function(id,i) {
+
+			let tmpTx={};
+			var tx = await this.arweave.transactions.get(id);
+			tmpTx['UnixTime']="0";
+			tx.get('tags').forEach(tag => {
+			let key = tag.get('name', {decode: true, string: true});
+			let value = tag.get('value', {decode: true, string: true});
+			if(key==="UnixTime") tmpTx['UnixTime']=value;
+
+			});
+
+			tmpTx['id']=id;
+			tmpTx['tx_status'] = await this.arweave.transactions.getStatus(id);
+			tmpTx['from'] = await arweave.wallets.ownerToAddress(tx.owner);
+			tmpTx['td_fee'] = arweave.ar.winstonToAr(tx.reward);
+			tmpTx['td_qty'] = arweave.ar.winstonToAr(tx.quantity);
+
+			mail_UnixTime.push(tmpTx);
+			displayMails(i);
```

Comment:
mails display could be refreshed only when all mails are received 
by calling displayMails  only when mail_UnixTime.length === mail_ids.length
 please let me know

Reply:
You can wait till all mails are downloaded the following way:

```
await Promise.all(mail_ids);
```

Reply:
it is  may be not needed to add an extra Promise,
since async  await does exactly what it is needed.
mail_ids already have all the ids and map does the iteration and fetch all the messages. 
displayMails only need to be triggered at the end of the map.

Reply:
The problem is it is triggered on every iteration, so if there are 1000 messages it will do 1000 sorts.

Reply:
no does not with the  IF test  on length